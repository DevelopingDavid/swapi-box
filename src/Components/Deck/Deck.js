import React from 'react'; 
import Card from '../Card/Card';

const Deck = (props) => {
  let cards;
  if(props.cardInfo.currentFilter === 'people') {
    cards = props.cardInfo.people.map(person => (
      <Card key={person.name} personInfo={person} state={props.cardInfo} />
    ));
  } else if (props.cardInfo.currentFilter === 'planets') {
    cards = props.cardInfo.planets.map(planet => (
      <Card key={planet.name} planetInfo={planet} state={props.cardInfo} />
    ));
  } else if (props.cardInfo.currentFilter === 'vehicles') {
    cards = props.cardInfo.vehicles.map(vehicle => (
      <Card key={vehicle.name} vehicleInfo={vehicle} state={props.cardInfo} />
    ));
  }

  return (
    <div>
      {cards}
    </div>
  )
}

export default Deck;