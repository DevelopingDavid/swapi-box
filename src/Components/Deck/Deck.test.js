import React from 'react';
import { shallow } from 'enzyme';
import Deck from './Deck';
const mockState = {
  people: [{name: "Luke Skywalker", homeworld: "Tatooine", population: "200000", species: "Human", language: "Galactic Basic"}]
}

describe('Deck', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Deck cardInfo={mockState}/>
    )
  });

  it('should create snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
});