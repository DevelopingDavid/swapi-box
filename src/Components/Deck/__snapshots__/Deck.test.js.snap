// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deck should create snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Deck
    cardInfo={
      Object {
        "people": Array [
          Object {
            "homeworld": "Tatooine",
            "language": "Galactic Basic",
            "name": "Luke Skywalker",
            "population": "200000",
            "species": "Human",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          personInfo={
            Object {
              "homeworld": "Tatooine",
              "language": "Galactic Basic",
              "name": "Luke Skywalker",
              "population": "200000",
              "species": "Human",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Luke Skywalker",
        "nodeType": "function",
        "props": Object {
          "personInfo": Object {
            "homeworld": "Tatooine",
            "language": "Galactic Basic",
            "name": "Luke Skywalker",
            "population": "200000",
            "species": "Human",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            personInfo={
              Object {
                "homeworld": "Tatooine",
                "language": "Galactic Basic",
                "name": "Luke Skywalker",
                "population": "200000",
                "species": "Human",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Luke Skywalker",
          "nodeType": "function",
          "props": Object {
            "personInfo": Object {
              "homeworld": "Tatooine",
              "language": "Galactic Basic",
              "name": "Luke Skywalker",
              "population": "200000",
              "species": "Human",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
